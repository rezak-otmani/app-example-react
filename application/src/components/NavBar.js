import React from "react";
import Scrollchor from 'react-scrollchor';

 
const NavBar = () => (
   
 
<div class="ui inverted segment">
 <div class="ui inverted secondary pointing stackable container massive menu">

    <Scrollchor to="" className="header item">Home</Scrollchor>
    <Scrollchor to="product" className="header item">Product</Scrollchor>
    <Scrollchor to="about" className="header item">About</Scrollchor>
    <Scrollchor to="featured" className="header item">Featured</Scrollchor>
    <Scrollchor to="contact" className="header item">Contact</Scrollchor>
    <Scrollchor to="demo" className="header item">Demo</Scrollchor>

</div>
</div>



)  

  
export default NavBar;
