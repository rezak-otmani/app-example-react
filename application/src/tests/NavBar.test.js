import React from 'react';
import Enzyme, { shallow, render, mount } from 'enzyme';
import renderer from 'react-test-renderer';
import NavBar from '../components/NavBar';


describe('NavBAr Component', () => {

  let wrapper;
 
  beforeEach(() => {
    wrapper = shallow(
      <NavBar />
    );
  });

        it('it should render NavBar ', () => {
   
              expect(wrapper.length).toBe(1);
		});

         it('should render all items for menu', () => {
        
             expect(wrapper.find('Scrollchor').length).toBe(6);
		});
        
         it('should have one active class item', () => {
             expect(wrapper.find('.active').length).toBe(1); 
          });
  
    describe('shows a links to scroll in the page', () => {


          it('shows a link to Home', () => {
             const link = wrapper.find({to: ''});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('Home')
           });
 
            it('shows a link to Product', () => {
             const link = wrapper.find({to: 'product'});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('Product')
           });

           it('shows a link to About', () => {
             const link = wrapper.find({to: 'about'});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('About')
           }); 

             it('shows a link to Featured', () => {
             const link = wrapper.find({to: 'featured'});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('Featured')
           });

                it('shows a link to Contact', () => {
             const link = wrapper.find({to: 'contact'});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('Contact')
           });

              it('shows a link to Demo', () => {
             const link = wrapper.find({to: 'demo'});
             expect(link.length).toEqual(1);
             expect(link.childAt(0).text()).toBe('Demo')
           });


});

              it('renders correctly', () => {
                const tree = renderer.create(
                 <NavBar />
                          ).toJSON();
                  expect(tree).toMatchSnapshot();
                });


});
