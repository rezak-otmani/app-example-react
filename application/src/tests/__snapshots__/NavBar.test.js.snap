// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[` 1`] = `
<div
  className="ui inverted segment"
>
  <div
    className="ui secondary stackable massive compact menu"
  >
    <a
      className="olive item active"
      href="#"
      onClick={[Function]}
    >
      Home
    </a>
    <a
      className="item"
      href="#product"
      onClick={[Function]}
    >
      Product
    </a>
    <a
      className="item"
      href="#about"
      onClick={[Function]}
    >
      About
    </a>
    <a
      className="item"
      href="#featured"
      onClick={[Function]}
    >
      Featured
    </a>
    <a
      className="item"
      href="#contact"
      onClick={[Function]}
    >
      Contact
    </a>
    <a
      className="item"
      href="#demo"
      onClick={[Function]}
    >
      Demo
    </a>
  </div>
</div>
`;
